sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,348 @@
+                }
+            }
+    }
+
+    /// Lists read receipts for the ChatThread.
+    /// - Parameters:
+    ///   - options: List chat read receipts options.
+    ///   - completionHandler: A completion handler that receives the list of read receipts on success.
+    public func listReadReceipts(
+        withOptions options: AzureCommunicationChatService.ListChatReadReceiptsOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<PagedCollection<ChatMessageReadReceipt>>
+    ) {
+        service.listChatReadReceipts(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(readReceipts):
+                completionHandler(.success(readReceipts), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a typing notification.
+    /// - Parameters:
+    ///    - options: Send typing notification options
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func sendTypingNotification(
+        withOptions options: AzureCommunicationChatService.SendTypingNotificationOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service.sendTypingNotification(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case .success:
+                completionHandler(.success(()), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a message to a ChatThread.
+    /// - Parameters:
+    ///    - chatMessage : Request for sending a message.
+    ///    - options: A list of options for the operation.
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func send(
+        message: SendChatMessageRequest,
+        withOptions options: AzureCommunicationChatService.SendChatMessageOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<SendChatMessageResult>
+    ) {
+        service.send(chatMessage: message, chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(sendMessageResult):
+                completionHandler(.success(sendMessageResult), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Gets a message by id.
+    /// - Parameters:
+    ///    - withId : The message id.
+    ///    - options: Get chat message options
+    ///    - completionHandler: A completion handler that receives the chat message on success.
+    public func getMessage(
+        withId chatMessageId: String,
```

Comment:
This is inconsistent with your update and delete methods. They should all be consistent.

Reply:
Updated